NeoMathEngine/src/GPU/CUDA/CudaDevice.h

```diff
@@ -37,22 +37,25 @@ struct CCudaDevice : public CCrtAllocatedObject {
 	int ThreadMaxCount;
 	dim3 ThreadMax3DCount;
 	int WarpSize;
-	void* Handles[CUDA_DEV_SLOT_COUNT];
+	void* Handle;
 
-	CCudaDevice( int deviceNumber, size_t memoryLimit );
+	CCudaDevice( int deviceNumber, size_t memoryLimit, void* handle );
 	~CCudaDevice();
+
+	CCudaDevice( const CCudaDevice& ) = delete;
+	CCudaDevice& operator=( const CCudaDevice& ) = delete;
 };
 
-// Checks if device slot is free.
-bool IsDeviceSlotFree( int deviceId, int slotIndex );
+// Returns the amount of captured slots on this device.
+int GetDeviceUsage( int deviceId );
 
-// Captures slot and returns it's hadnle.
-// Handle must be released after work (ReleaseDeviceSlot).
-// Returns nullptr if slot is busy.
-void* CaptureDeviceSlot( int deviceId, int slotIndex, bool reuse );
+// Captures slots and returns it's hadnle.
+// Handle must be released after work (ReleaseDeviceSlots).
+// Returns nullptr if device is too busy.
+void* CaptureDeviceSlots( int deviceId, int slotCount );
```

Comment:
If we use CreateDevice we will able to delete CaptureDeviceSlots from .h.

Reply:
Fixed